src/wp-includes/fonts/class-wp-font-collection.php

```diff
@@ -187,7 +189,34 @@ private function sanitize_and_validate_data( $data ) {
 		$schema = self::get_sanitization_schema();
 		$data   = WP_Font_Utils::sanitize_from_schema( $data, $schema );
 
-		$required_properties = array( 'name', 'font_families' );
+		if ( ! isset( $data['src'] ) && ! isset( $data['font_families'] ) ) {
+			$message = sprintf(
+				// translators: 1: src. 2: font_families.
+				__( 'Font collection must provide either "%1$s" or "%2$s".' ),
+				'src',
+				'font_families'
+			);
+			_doing_it_wrong( __METHOD__, $message, '6.5.0' );
+			return new WP_Error( 'font_collection_missing_property', $message );
+		}
+
+		if ( isset( $data['src'], $data['font_families'] ) ) {
+			$message = sprintf(
+			// translators: 1: src. 2: font_families.
+				__( 'Font collection must only provide "%2$s" or "%2$s", not both.' ),
+				'src',
+				'font_families'
+			);
+			_doing_it_wrong( __METHOD__, $message, '6.5.0' );
+			return new WP_Error( 'font_collection_duplicate_property', $message );
+		}
```

Comment:
Should check the same around `src` and `categories`?

Reply:
Yes, makes sense. Missed that one.

Reply:
Thinking twice, we should remove this check because that's the use case we will have in the Google Fonts collections.

```
wp_register_font_collection(
	'google-fonts',
	array(
		'src'         => 'https://s.w.org/images/fonts/17.7/collections/google-fonts-with-preview.json',
		'name'        => __( 'Google Fonts' ),
		'description' => __( 'Install from Google Fonts. Fonts are copied to and served from your site.' ),
		'categories'  => $categories,
	)
);
```